crengine/src/hist.cpp

```diff
@@ -516,6 +526,7 @@ ldomXPointer CRFileHist::restorePosition( ldomDocument * doc, lString16 fpathnam
     int index = findEntry( name, path, (lvsize_t)sz );
     if ( index>=0 ) {
         makeTop( index );
+        gDOMVersionRequested = _records[0]->getDOMversion();
         return doc->createXPointer( _records[0]->getLastPos()->getStartPos() );
```

Comment:
I wonder if that is not too late (the DOM might already be built if you are calling doc->createXPointer(), and so that old saved value wouldn't have any effect ?)
Also, given that you skipped all the intermediate DOM_VERSIONs and we are hopefully done with that work and there won't be any new one :), once this is merged, you have only 2 cases to handle:
- books with old hist files without yet any dom_version saved: you should request a very old one (we use `20171225 -- arbitrary day in the past`), and save that to the hist file, so it's used with next openings
- newly opened books with no hist file: use the current dom version, so they get all the good/fixed dom stuff

Note: the need for that DOM version is only because some nodes are switched from/to `display: block` to/from `display: inline` - and this may cause some of them to be (or no more be) wrapped in `<autoBoxing>` elements (anonymous block container for inline nodes that were at the same level of other block nodes).
That might be rare with well-formed books - but it may happen with our change to list item handling (that are now a proper block element, while there were inline previously and may have been wrapped in a `<autoBoxing>`  if there were `<P>` siblings for example): so you may have chances to see and test for highlights survival or disappearance by highlighting text inside lists.